src/frequenz/channels/utils/file_watcher.py

```diff
@@ -52,6 +53,7 @@ def __init__(
                 and pathlib.Path(path_str).is_file()
             ),
         )
+        self._changes: Set[FileChange] = set()
```

Comment:
It looks like ordering is important here and I don't think `set` preserves order. Also, why do we want to compress `FileChange`s that are the same, suppose we have this sequence:

1. `/a` is added
2. `/a` is deleted
3. `/a` is added again

With this implementation we could only get one add and one deletion so we would think the file doesn't exist but it does.

If this introduced a bug that wasn't detected by test, we should probably add a test for this too :)

Reply:
OK, this turned out to be invalid, because `awatch` returns a `set`, so there is no order anyway. It is kind of weird that you could potentially get unordered changes on the same file in the set. Maybe the underlying library only accumulates one change per file in every loop, but I couldn't find any mention of this neither in the Python or the Rust projects.